### DISCUSSION TITLE
Weekly Report 19.04 - 25.04

### DISCUSSION THREAD

**Discussion by Umbranecis** [SE-TINF22B6/Plapy#129] [Weekly reports]
Greetings,

This week we worked to further improve our code performance and data traffic.
To do this, we
- moved the database requests to only be called when the objects are actually needed, instead of generating all objects from the database in runtime.
- ensured the database structure gets automatically updated by typescript and typeORM annotations.

Further more, we created a database on a server, so we can now release all the database changes on the live bot.

We also created commands that give you the chance to show all playlists that were created on a server.

To make sure our application is stateful, we also save the server_id in the database and filter by server id, so you only have access to playlists on your current server.
In future steps, we might also make them user specific, but that remains to be discussed.

We also started learning react to be able to properly code the frontend page we want to use to easier controll the bot.

**Comment by nuo-li** [comment]
Hello Plapy team, could you please explain more about how to automatically update the database structure by typescript and typeORM annotations? Thanks!
